File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        vector<int> arr;
        for(int i=0;i<nums.size();i++){
            if(nums[abs(nums[i])-1]<0){
                 arr.push_back(abs(nums[i]));
            }
            else{
              nums[abs(nums[i])-1]*=-1;
            }
        }
        return arr;
    }
};

File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.java
import java.io.*;
import java.util.*;
class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        List<Integer> arr=new ArrayList<>();
        for(int i=0;i<nums.length;i++){
            if(nums[Math.abs(nums[i])-1]<0){
                 arr.add(Math.abs(nums[i]));
            }
            else{
              nums[Math.abs(nums[i])-1]*=-1;
            }
        }
        return arr;
    }
}